### ISSUE TITLE
Input mode automatically exits after pressing a single key

### ISSUE THREAD

**Issue by Pound-Hash** [qutebrowser/qutebrowser#7436]
**Version info**:

qutebrowser v2.5.2
Git commit: 
Backend: QtWebEngine 5.15.10, based on Chromium 87.0.4280.144
Qt: 5.15.6

**Does the bug happen if you start with `--temp-basedir`?**:
No

**Description**
As reported in issue #3596, insert mode is automatically exiting after a single key press. This happens, for example, at: https://pkgstats.archlinux.de/packages?query=a

I tried toggling the settings `input.insert_mode.leave_on_load` & `hints.leave_on_load` to no avail.

**How to reproduce**
1. Enter insert mode (i)
2. Press a key
3. Insert mode exits

**Comment by Pound-Hash** [comment]
I've resolved the issue. The problem lies with **two** settings: `input.mode_override` and `input.insert_mode.leave_on_load`.

The strange thing is that `input.mode_override` didn't have **any** values selected from the settings page (as seen in this screenshot).
![image](https://user-images.githubusercontent.com/51193876/196064938-8e21d6da-a71a-4b60-8529-8bb891aa8d02.png)

It's the only setting like this. When I saw it yesterday, I figured I'd better select a value. But the only way to resolve this issue was to delete that setting from my autoconfig.yml manually, which I know I'm not supposed to mess with. But clicking on the values in that setting do **not** deselect them, so there wasn't any other option. Finally, I also had to set `input.insert_mode.leave_on_load` to `false`. Both steps must be done for the issue to resolve.

**Comment by Pound-Hash** [comment]
I'm not sure where to add this, but I now see that the setting `input.insert_mode.leave_on_load` can be set to `True` and the issue is still resolved. The problem here was that I needed to restart qutebrowser after making the changes to `input.mode_override`.

**Comment by The-Compiler** [comment]
I removed that setting from the GUI part, as it makes no sense for it to be set globally anyways.

**Comment by The-Compiler** [comment]
Whoops, that breaks tests because now those can't set it anymore either :sweat_smile:  Will need to take a closer look at this somewhen later.

**Comment by toofar** [comment]
The `no_autoconfig` options doesn't just hide options from the :set web interface (which doesn't understand url patterns) but also prevents them from being set via `:set -u example.com input.mode_override passthrough`, which otherwise works fine. Personally I use the :set command in preference to config.py and I would rather settings weren't removed from that interface without very good justification.

I think the two point five difficulties here are:

1. it's easy to set this option globally when you very likely don't want to do that (probably we should either add a disallow_globally option or some special case validation to this option)
2. the default option is `null` but you can't set that via the :set web interface
    * I think this only applies for radio options with `none_ok: true`? You seem to be able to set string and list values to empty just fine, not sure if there are any dicts that allow none
3. (actually 2.5) you can't :unset options from the :set web interface

Here's a simplistic patch to support setting this option back to None, I suspect it works because `''` is already the way to set things to null via :set. But adding `''` as a valid value kinda makes a mockery of the `none_ok` option. Probably we should make the web interface understand that option. I think first a bit more discovery is required to make sure special casing is only required for radio options:

```patch

diff --git i/qutebrowser/config/configdata.yml w/qutebrowser/config/configdata.yml
index 6600399bd78d..ef9edd1449ea 100644
--- i/qutebrowser/config/configdata.yml
+++ w/qutebrowser/config/configdata.yml
@@ -1914,6 +1914,7 @@ input.mode_override:
       - normal
       - insert
       - passthrough
+      - ''
   supports_pattern: true
   desc: Mode to change to when focusing on a tab/URL changes.
 
diff --git i/qutebrowser/html/settings.html w/qutebrowser/html/settings.html
index f89aaa610b5f..5ddad9cd2a8a 100644
--- i/qutebrowser/html/settings.html
+++ w/qutebrowser/html/settings.html
@@ -191,7 +191,7 @@ summary::selection {
                   checked
                 {% endif %}>
               <label for="input-{{ option.name }}-{{ loop.index0 }}">
-                {{ value }}
+                {{ 'null' if value == '' else value }}
               </label>
               </div>
           {% endfor %}
           
```

**Comment by The-Compiler** [comment]
Right, I admittedly jumped the gun a bit there, apologies. Agreed that setting it via `:set` should still be possible, and agreed that we probably need some way to prevent it from being set globally.
